programs/perpetuals/src/instructions/remove_liquid_stake.rs

```diff
@@ -0,0 +1,321 @@
+                ctx.accounts.governance_program.to_account_info(),
+                params.amount,
+            )?;
+        }
+
+        // apply delta to user stake
+        staking.liquid_stake.amount =
+            math::checked_sub(staking.liquid_stake.amount, params.amount)?;
+
+        // Apply delta to current and next round
+        {
+            // forfeit current round participation, if any
+            if staking
+                .liquid_stake
+                .qualifies_for_rewards_from(&cortex.current_staking_round)
+            {
+                // overlap
+                if staking.liquid_stake.overlap_amount > 0
+                    && staking.liquid_stake.overlap_time >= cortex.current_staking_round.start_time
+                {
+                    // In case of overlap, takes overlapped tokens first (last tokens put in staking)
+                    //
+                    // if there are not enough tokens, takes it up from long lasting staked tokens reserve
+                    if params.amount > staking.liquid_stake.overlap_amount {
+                        staking.liquid_stake.overlap_amount = 0;
+
+                        cortex.current_staking_round.total_stake = math::checked_sub(
+                            cortex.current_staking_round.total_stake,
+                            math::checked_sub(params.amount, staking.liquid_stake.overlap_amount)?,
+                        )?;
+                    } else {
+                        staking.liquid_stake.overlap_amount =
+                            math::checked_sub(staking.liquid_stake.overlap_amount, params.amount)?;
+
+                        cortex.current_staking_round.total_stake = math::checked_sub(
+                            cortex.current_staking_round.total_stake,
+                            math::checked_sub(params.amount, staking.liquid_stake.overlap_amount)?,
+                        )?;
+                    }
+                } else {
+                    cortex.current_staking_round.total_stake =
+                        math::checked_sub(cortex.current_staking_round.total_stake, params.amount)?;
+                }
+            }
+
+            // apply delta to next round
+            cortex.next_staking_round.total_stake =
+                math::checked_sub(cortex.next_staking_round.total_stake, params.amount)?;
+        }
+    };
+
+    // Unstake owner's tokens
+    {
+        msg!("Transfer tokens");
+        let perpetuals = ctx.accounts.perpetuals.as_mut();
+
+        perpetuals.transfer_tokens(
+            ctx.accounts.staking_token_account.to_account_info(),
+            ctx.accounts.lm_token_account.to_account_info(),
+            ctx.accounts.transfer_authority.to_account_info(),
+            ctx.accounts.token_program.to_account_info(),
+            params.amount,
+        )?;
+    }
+
+    // pause auto-claim if there are no more staked token,
+    {
+        if !ctx.accounts.stakes_claim_cron_thread.paused
```

Comment:
note: so no clean of the claim thread for now (and refund). That's fine keep for later